core/modules/xrdsvc/SsiProvider.cc

```diff
@@ -0,0 +1,160 @@
+// object. The SSI framework looks for these symbols when the shared library
+// plug-in is loaded. The framework must find a valid provider object at load
+// time or it will refuse to use the shared library. As the library is never
+// unloaded, the object does not need to be deleted.
+//
+XrdSsiProvider *XrdSsiProviderServer = 
+                new  lsst::qserv::xrdsvc::SsiProviderServer;
+
+XrdSsiProvider *XrdSsiProviderLookup = XrdSsiProviderServer;
+  
+/******************************************************************************/
+/*                            D e s t r u c t o r                             */
+/******************************************************************************/
+  
+namespace lsst {
+namespace qserv {
+namespace xrdsvc {
+
+SsiProviderServer::~SsiProviderServer() {}
+  
+/******************************************************************************/
+/*                                  I n i t                                   */
+/******************************************************************************/
+
+bool SsiProviderServer::Init(XrdSsiLogger* logP,  XrdSsiCluster* clsP,
+                             char const*   cfgFn, char const*    parms,
+                             int           argc,  char**         argv) {
+
+    lsst::qserv::xrdsvc::XrdName x;
+
+    // Establish our instance name (many different qservs may be running here).
+    //
+    _name = x.getName();
+
+    // Save the ssi logger as it places messages in anoher file than our log.
+    //
+    _logSsi = logP;
+
+    // Save the cluster object as we will need to use it to inform the cluster
+    // when chunks come and go. We also can use it to schedule ourselves. The
+    // object or its absence will indicate whether or not we need to provide
+    // any service other than QueryResource().
+    //
+    _cmsSsi = clsP;
+
+    // We would process the configuration file (if present), any present
+    // parameters and the command line arguments. However, at the moment, we
+    // have nothing of interest in any of these arguments. So, we ignore them.
+    //
+
+    // Herald our initialization
+    //
+    LOG_INFO("SsiProvider initializing...");
+    _logSsi->Msg("Qserv", "Provider Initializing");
+
+    // Initialize the inventory. We need to be able to handle QueryResource()
+    // calls either in the data provider and the metadata provider (we can be
+    // either one).
+    //
+    _chunkInventory.reset(new wpublish::ChunkInventory(_name));
+
+    // If we are a data provider (i.e. xrootd) then we need to get the service
+    // object. It will print the exported paths. Otherwise, we need to print
+    // them here. This is kludgy and should be corrected when we transition to a
+    // single shared memory inventory object which should do this by itself.
+    //
+    if(clsP && clsP->DataContext()) _service.reset(new SsiService(logP));
+      else {
```

Comment:
The indentation is confusing here, I'd make it just:

    if(clsP && clsP->DataContext()) {
        _service.reset(new SsiService(logP));
    } else {